tar_scm.py

```diff
@@ -727,33 +727,41 @@ def write_changes(changes_filename, changes, version, author):
     shutil.move(tmp_fp.name, changes_filename)
 
 
-def detect_changes_commands_git(repodir, changes):
+def detect_changes_commands_git(repodir, subdir, changes):
     """Detect changes between GIT revisions."""
     last_rev = changes['revision']
 
     if last_rev is None:
-        last_rev = safe_run(['git', 'log', '-n1', '--pretty=format:%H',
-                             '--skip=10'], cwd=repodir)[1]
-    current_rev = safe_run(['git', 'log', '-n1', '--pretty=format:%H'],
-                           cwd=repodir)[1]
+        last_cmd = ['git', 'log', '-n1', '--pretty=format:%H', '--skip=10']
+        if subdir:
+            last_cmd += ['--', subdir]
+        last_rev = safe_run(last_cmd, cwd=repodir)[1]
+    current_cmd = ['git', 'log', '-n1', '--pretty=format:%H']
+    if subdir:
+        current_cmd += ['--', subdir]
+    current_rev = safe_run(current_cmd, cwd=repodir)[1]
 
     if last_rev == current_rev:
         logging.debug("No new commits, skipping changes file generation")
         return
 
-    logging.debug("Generating changes between %s and %s", last_rev,
-                  current_rev)
+    dbg_msg = "Generating changes between %s and %s" % (last_rev, current_rev)
+    if subdir:
+        dbg_msg += " (for subdir: %s)" % (subdir)
+    logging.debug(dbg_msg)
 
-    lines = safe_run(['git', 'log',
-                      '--reverse', '--no-merges', '--pretty=format:%s',
-                      "%s..%s" % (last_rev, current_rev)], repodir)[1]
+    cmd = ['git', 'log', '--reverse', '--no-merges', '--pretty=format:%s',
+           "%s..%s" % (last_rev, current_rev)]
+    if subdir:
```

Comment:
This is the third time in this function that this logic is duplicated, so I think it's justified creating a helper function which constructs the arguments to the `git log` command (whilst respecting `subdir`) and then runs the command.  That will help shrink this function which is starting to get uncomfortably long.